src/future/batch_semaphore.rs

```diff
@@ -447,6 +492,15 @@ impl BatchSemaphore {
 
         state.waiters.remove(index).unwrap();
         assert!(waiter.is_queued.swap(false, Ordering::SeqCst));
+
+        if matches!(self.fairness, Fairness::StrictlyFair) && index == 0 {
```

Comment:
Nit-ish: I don't think `matches!` is well suited here. We want to be forced to have a look at this if we change `Fairness`.

Reply:
Yeah, updated to use `match`